packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/RangeBondLongShortPairFinancialProductLibrary.sol

```diff
@@ -63,6 +63,7 @@ contract RangeBondLongShortPairFinancialProductLibrary is LongShortPairFinancial
         require(highPriceRange > lowPriceRange, "Invalid bounds");
 
         RangeBondLongShortPairParameters memory params = longShortPairParameters[LongShortPair];
+        require(params.highPriceRange == 0 && params.lowPriceRange == 0, "Parameters already set");
```

Comment:
this is silly as we already had declared the `params` object to validate this but did not have the require!

Reply:
What if someone sets the `lowPriceRrange = 0`?

Reply:
well `highPriceRange > lowPriceRange` from the requirements on line 63 so it would not be possible for them both to be 0. you _could_ set `lowPriceRrange = 0` but by definition `highPriceRange` could then also not be zero and this requirement would fail. i.e the only case where this require passes is when there is nothing set for this value.

do you agree?

Reply:
good point, I agree.